crates/runtime/src/dataconnector/github/mod.rs

```diff
@@ -502,13 +557,32 @@ pub(crate) fn filter_pushdown(expr: &Expr) -> FilterPushdownResult {
                 };
             }
 
-            let value = if let ScalarValue::Utf8(Some(v)) = &value {
-                match column.name.as_str() {
-                    "state" => ScalarValue::Utf8(Some(v.to_lowercase())),
-                    _ => value.clone(),
+            let value = match value {
+                ScalarValue::Utf8(Some(v)) => {
+                    if column.name == "state" {
+                        if v.to_lowercase() == "merged" {
```

Comment:
@peasee  - why we change from `merged` to `closed` here?

Reply:
GitHub search combines `merged` and `closed` pull requests under the same search term for `closed`. If you need to find merged PRs, the inexact filter will also return closed PRs which leaves the in-memory table to perform the final filter.